js/src/main/kotlin/org/rust/js/RsWasmBindgenLineMarkerProvider.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.js
+
+import com.intellij.codeHighlighting.Pass
+import com.intellij.codeInsight.daemon.DefaultGutterIconNavigationHandler
+import com.intellij.codeInsight.daemon.RelatedItemLineMarkerInfo
+import com.intellij.codeInsight.daemon.RelatedItemLineMarkerProvider
+import com.intellij.lang.javascript.psi.JSElement
+import com.intellij.lang.javascript.psi.ecma6.TypeScriptClass
+import com.intellij.lang.javascript.psi.ecma6.TypeScriptEnum
+import com.intellij.lang.javascript.psi.ecma6.TypeScriptFunction
+import com.intellij.navigation.GotoRelatedItem
+import com.intellij.openapi.editor.markup.GutterIconRenderer
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
+import com.intellij.psi.util.findDescendantOfType
+import com.intellij.psi.util.nextLeaf
+import com.intellij.util.ConstantFunction
+import icons.JavaScriptPsiIcons
+import org.rust.lang.core.crate.crateGraph
+import org.rust.lang.core.psi.RsEnumItem
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.RsStructItem
+import org.rust.lang.core.psi.ext.RsOuterAttributeOwner
+import org.rust.lang.core.psi.ext.RsVisibilityOwner
+import org.rust.lang.core.psi.ext.containingCargoPackage
+import org.rust.lang.core.psi.ext.findOuterAttr
+import org.rust.lang.core.psi.impl.RsBaseTypeImpl
+import java.nio.file.Paths
+
+class RsWasmBindgenLineMarkerProvider : RelatedItemLineMarkerProvider() {
+    override fun getLineMarkerInfo(element: PsiElement): RelatedItemLineMarkerInfo<*>? {
+        // Line marker will be available only if there is wasm-bindgen dependency in the project,
+        // current element is public and can have outer attribute
```

Comment:
AFAIU currently we only support `RsFunction`, `RsEnumItem`, `RsStructItem`, and `RsImplItem` as `element`. So maybe it's better to restrict the scope here?
I mean check that element is `RsFunction` or ... or `RsImplItem` instead of checking `element !is RsOuterAttributeOwner`, `element is RsVisibilityOwner`, and so forth. It might be more explicit and easy to understand.